src/Microsoft.DotNet.Interactive.Formatting/TabularData/TabularDataFormatterSource.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Html;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive.Formatting.TabularData
+{
+    internal class TabularDataFormatterSource : ITypeFormatterSource
+    {
+        public IEnumerable<ITypeFormatter> CreateTypeFormatters()
+        {
+            yield return new HtmlFormatter<TabularDataResource>((value, context) =>
+            {
+                IReadOnlyList<IHtmlContent> headers =
+                    value.Schema
+                         .Fields
+                         .Select(f => (IHtmlContent) td(span(f.Name)))
+                         .ToArray();
+
+                IReadOnlyList<IHtmlContent> rows =
+                    value.Data
+                         .Select(d => (IHtmlContent) tr(d.Values.Select(v => td(v))))
```

Comment:
How do we ensure that the values are preserving the order of the column names?

Reply:
Nice catch!

Updating the test to capture this.

Reply:
Actually I think the answer here is we don't. They're expected to be in the same order, otherwise we go down a kind of unfortunate perf path.